bin/runtime/src/lib.rs

```diff
@@ -767,8 +766,168 @@ pub type Executive = frame_executive::Executive<
     frame_system::ChainContext<Runtime>,
     Runtime,
     AllPalletsWithSystem,
+    (
+        /* pallet staking migrations: from V7 to V12 */
+        original_staking_migrations::StakingBagsListMigrationV8,
+        // this requires custom migrations for sake of try-runtime post_upgrade hook
+        // that has a bug for blockchains that do not use BagsList
+        custom_staking_migration::BumpStorageVersionFromV8ToV9<Runtime>,
+        pallet_staking::migrations::v10::MigrateToV10<Runtime>,
+        pallet_staking::migrations::v11::MigrateToV11<
+            Runtime,
+            // this argument is different as original one named VoterList referred to pallet_bags_list
+            original_staking_migrations::StakingMigrationV11OldPallet,
+            original_staking_migrations::StakingMigrationV11OldPallet,
+        >,
+        pallet_staking::migrations::v12::MigrateToV12<Runtime>,
+    ),
 >;
 
+mod original_staking_migrations {
```

Comment:
could you move this code to migration.rs?

Reply:
done